workspaces/diff-engine/src/events/shape.rs

```diff
@@ -214,3 +219,130 @@ impl Event for FieldRemoved {
     "FieldRemoved"
   }
 }
+
+impl From<ShapeAdded> for ShapeEvent {
+  fn from(event: ShapeAdded) -> Self {
+    Self::ShapeAdded(event)
+  }
+}
+
+impl From<BaseShapeSet> for ShapeEvent {
+  fn from(event: BaseShapeSet) -> Self {
+    Self::BaseShapeSet(event)
+  }
+}
+
+impl From<FieldAdded> for ShapeEvent {
+  fn from(event: FieldAdded) -> Self {
+    Self::FieldAdded(event)
+  }
+}
+
+impl From<FieldShapeSet> for ShapeEvent {
+  fn from(event: FieldShapeSet) -> Self {
+    Self::FieldShapeSet(event)
+  }
+}
+
+impl From<ShapeParameterAdded> for ShapeEvent {
+  fn from(event: ShapeParameterAdded) -> Self {
+    Self::ShapeParameterAdded(event)
+  }
+}
+
+impl From<ShapeParameterShapeSet> for ShapeEvent {
+  fn from(event: ShapeParameterShapeSet) -> Self {
+    Self::ShapeParameterShapeSet(event)
+  }
+}
+
+// Conversions from commands
+// -------------------------
+
+impl From<ShapeCommand> for ShapeEvent {
+  fn from(shape_command: ShapeCommand) -> Self {
+    match shape_command {
+      ShapeCommand::AddShape(command) => ShapeEvent::from(ShapeAdded::from(command)),
+      ShapeCommand::SetBaseShape(command) => ShapeEvent::from(BaseShapeSet::from(command)),
+      ShapeCommand::AddField(command) => ShapeEvent::from(FieldAdded::from(command)),
+      ShapeCommand::AddShapeParameter(command) => {
+        ShapeEvent::from(ShapeParameterAdded::from(command))
+      }
+      ShapeCommand::SetParameterShape(command) => {
+        ShapeEvent::from(ShapeParameterShapeSet::from(command))
+      }
+      _ => unimplemented!(
+        "conversion from shape command to shape event not implemented for variant: {:?}",
+        shape_command
+      ),
+    }
+  }
+}
+
+impl From<shape_commands::AddShape> for ShapeAdded {
+  fn from(command: shape_commands::AddShape) -> Self {
+    Self {
+      shape_id: command.shape_id,
+      base_shape_id: command.base_shape_id,
+      name: command.name,
+      parameters: ShapeParametersDescriptor::empty_dynamic(),
+      event_context: None,
```

Comment:
What is the plan for CommandContext/EventContext? We will definitely need something like it soon

Reply:
Plan is to land this PR so we can then deal with details like this in parallel. More specifically for this case: accept `CommandContext` as part of commands coming from the UI (or generate our own for commands generated from the Rust domain itself), then map them near 1:1 to `EventContext` as part of the command -> event conversion.